hack/vendor.sh

```diff
@@ -19,7 +19,12 @@ case $# in
 	;;
 # If user passed arguments to the script
 1)
-	eval "$(grep -E "^clone [^ ]+ $1" "$0")"
+	path=$(pwd)"/hack/vendor.sh"
+	if ! cloneGrep="$(grep -E "^clone [^ ]+ $1" $path)"; then
+    	echo >&2 "error: failed to find 'clone ... $1' in $0"
+    	exit 1
```

Comment:
"$BASH_SOURCE" will always refer to the current file in Bash. Totally agree.

Here is my reason why not choose $BASH_SOURCE, correct me if there is something wrong.😄
In the head of vendor.sh, there is `cd "$(dirname "$BASH_SOURCE")/.."`, and after this command the current dir will be `.../github/docker/docker`, right?

Then the grep Command `grep -E "^clone [^ ]+ $1" "$BASH_SOURCE"` will try to work under `.../github/docker/docker`. While `$BASH_SOURCE` will be `./vendor.sh`, not in `.../github/docker/docker`, but in `.../github/docker/docker/hack`. As a result, the grep command will always fail with the error `grep: ./vendor.sh: No such file or directory`.